docs/csharp/language-reference/operators/equality-comparison-operator.md

```diff
@@ -11,16 +10,46 @@ helpviewer_keywords:
 ms.assetid: 34c6b597-caa2-4855-a7cd-38ecdd11bd07
 ---
 # == Operator (C# Reference)
-For predefined value types, the equality operator (`==`) returns true if the values of its operands are equal, `false` otherwise. For reference types other than [string](../../../csharp/language-reference/keywords/string.md), `==` returns `true` if its two operands refer to the same object. For the `string` type, `==` compares the values of the strings.  
-  
-## Remarks  
- User-defined value types can overload the `==` operator (see [operator](../../../csharp/language-reference/keywords/operator.md)). So can user-defined reference types, although by default `==` behaves as described above for both predefined and user-defined reference types. If `==` is overloaded, [!=](../../../csharp/language-reference/operators/not-equal-operator.md) must also be overloaded. Operations on integral types are generally allowed on enumeration.  
-  
-## Example  
- [!code-csharp[csRefOperators#36](../../../csharp/language-reference/operators/codesnippet/CSharp/equality-comparison-operator_1.cs)]  
-  
-## See Also
-
-- [C# Reference](../../../csharp/language-reference/index.md)  
-- [C# Programming Guide](../../../csharp/programming-guide/index.md)  
-- [C# Operators](../../../csharp/language-reference/operators/index.md)
+
+The equality operator `==` returns `true` if its operands are equal, `false` otherwise.
+
+## Value types equality
+
+Operands of the [built-in value types](../keywords/value-types-table.md) are equal if their values are equal:
+
+[!code-csharp-interactive[value types equality](~/samples/snippets/csharp/language-reference/operators/EqualityAndNonEqualityExamples.cs#ValueTypesEquality)]
+
+By default, the `==` operator is not defined for a user-defined [struct](../keywords/struct.md) type. A user-defined type can [overload](#operator-overloadability) the `==` operator.
```

Comment:
This should state that `==` is defined for enum types. That slipped through the cracks in the new (more clear) description.

Reply:
@BillWagner indeed, as behavior for `struct`s is described, `enum` case also should be stated explicitly. Thank you for the remark. I've added the clarification, please check the latest commit.

Reply:
I'll suggest this small word change:

"Two operands of the same [enum](../keywords/enum.md) type are equal if the ..."

It's a small change, but the text you have could be misinterpreted to support something like the following:

```csharp
    enum Color
    {
        Red,
        Green,
        Blue
    }

    enum Status
    {
        Red,
        Yellow,
        Green
    }

 bool equal = (Status.Red == Color.Red); // CS0019 (operator == cannot be applied to ...)
```

Reply:
@BillWagner that's a good catch! Thanks. Fixed.